analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -450,6 +463,425 @@ public void Finally_NestedFinally()
                 "InInnerFinally",
                 "AfterInnerFinally",
                 "AfterOuterFinally");
+
+            validator.TagStates("InOuterFinally").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("InInnerFinally").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("InInnerTry").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("AfterInnerFinally").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasNoException(x));
+
+            validator.TagStates("AfterOuterFinally").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void TryCatch_ThrowInTry_SingleCatchBlock()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+    throw new System.Exception();
+    tag = ""UnreachableInFinally"";
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",          // With Exception thrown by Tag("InTry")
+                "InCatch",          // With Exception thrown by `throw`
+                "AfterCatch",
+                "AfterCatch");      // Should go away after https://github.com/SonarSource/sonar-dotnet/pull/5745 is done.
+
+            validator.TagStates("InCatch").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasUnknownException(x))
+                     .And.ContainSingle(x => HasExceptionOfTypeException(x));
+
+            validator.TagStates("AfterCatch").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasUnknownException(x))
+                     .And.ContainSingle(x => HasExceptionOfTypeException(x));
+        }
+
+        [TestMethod]
+        public void TryCatch_ThrowInTry_SingleCatchBlock_ReThrow()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+    throw new System.Exception();
+    tag = ""UnreachableInFinally"";
+}
+catch
+{
+    tag = ""InCatch"";
+    throw;
+}
+tag = ""UnreachableAfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",          // With Exception thrown by Tag("InTry")
+                "InCatch");         // With Exception thrown by `throw`
+
```

Comment:
In this specific test (not others), let's assert `validator.ValidateExitReachCount(1)`, only for the exception flow.

It would be actually nice to update `ValidatorTestCheck` to replace `exitReachedCount++` with `ExitStates.Add(context.State)` and be able to assert the exist states. And then 
```
validator.ExitStates.Should().HaveCount(2)
                     .And.ContainSingle(x => HasUnknownException(x))
                     .And.ContainSingle(x => HasExceptionOfTypeException(x));
```

Reply:
This assertion would be unique and doesn't deserve a special method

Reply:
exit reach is 0 now. Have to check why

Reply:
Found out that `rethrow` has its own semantics. Additionally, after `throw`, if there are no successors, we need to call exit reached.